ConsoleAppAzureServices/ConsoleAppAzureServices/Program.cs

```diff
@@ -1,47 +1,43 @@
 ﻿using System;
 using System.Threading.Tasks;
-using Microsoft.Azure.KeyVault;
-using Microsoft.Azure.Management.ResourceManager;
-using Microsoft.Azure.Services.AppAuthentication;
-using Microsoft.Rest;
+using Azure;
+using Azure.Identity;
+using Azure.ResourceManager.Resources;
+using Azure.ResourceManager.Resources.Models;
+using Azure.Security.KeyVault.Secrets;
 
 namespace ConsoleAppAzureServices
 {
     class Program
     {
-        static void Main()
+        static async Task Main()
         {
-            AzureServiceTokenProvider azureServiceTokenProvider = new AzureServiceTokenProvider();
+            DefaultAzureCredential credential = new DefaultAzureCredential();
 
-            GetSecretFromKeyVault(azureServiceTokenProvider).Wait();
+            await GetSecretFromKeyVault(credential);
 
-            GetResourceGroups(azureServiceTokenProvider).Wait();
-
-            if (azureServiceTokenProvider.PrincipalUsed != null)
-            {
-                Console.WriteLine($"{Environment.NewLine}Principal used: {azureServiceTokenProvider.PrincipalUsed}");
-            }
+            await GetResourceGroups(credential);
 
+            Console.WriteLine($"{Environment.NewLine}Principal used:{credential}");
             Console.ReadLine();
         }
 
-        private static async Task GetSecretFromKeyVault(AzureServiceTokenProvider azureServiceTokenProvider)
+        private static async Task GetSecretFromKeyVault(DefaultAzureCredential credential)
         {
-            KeyVaultClient keyVaultClient =
-                new KeyVaultClient(
-                    new KeyVaultClient.AuthenticationCallback(azureServiceTokenProvider.KeyVaultTokenCallback));
+            SecretClient secretClient =
+                new SecretClient(new Uri("https://{keyVaultName}.vault.azure.net/"), credential);
 
             Console.WriteLine("Please enter the key vault name");
 
-            var keyVaultName = Console.ReadLine();
+            Console.ReadLine();
```

Comment:
You're using a placeholder in the URI before even prompting for it, though in prompting for it you never actually assign to a variable that they can use. Instead, you might do something like this (and capitalize "Key Vault" since it' s a product name):
```c#
Console.WriteLine("Please enter the Key Vault name");
string keyVaultName = Console.ReadLine();

SecretrClient secretClient = new SecretClient(new Uri($"https://{keyVaultName}.vault.azure.net"), credential);
```

Make sure you run the samples before hand. Line 36 below would've failed since "{keyVaultName}.vault.azure.net" is invalid nor exists.